clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -354,6 +709,51 @@ dpct_memcpy(pitched_data src, const sycl::id<3> &src_id,
 }
 } // namespace detail
 
+template <typename T>
+bool check_duplicate_entries(int count, T **entries) {
+  std::unordered_set<T *> uniqueEntries;
+
+  for (size_t i = 0; i < count; ++i) {
+    auto result = uniqueEntries.insert(entries[i]);
+    if (!result.second) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+static inline void map_resources(int count, interop_mem_wrapper **handles,
```

Comment:
Why we need `static` in a header file?

Reply:
`static` keyword is added to prevent multiple redefinitions of the function when compiling multiple src files

If this is not a good enough reason, please let me know.

Reply:
`inline` will get rid of linking errors.